Python/Product/PythonTools/SharedProject/Debugger/ProvideDebugAdapterAttribute.cs

```diff
@@ -0,0 +1,129 @@
+             *     - Set the "Attach" property to "1" below
+             *     - Provide a port supplier GUID.  To attach to processes on the local machine by PID, the default
+             *         port supplier is suffient, and can be used by uncommenting the "PortSupplier" property below.
+             *     - Provide a custom IAdapterLauncher implementation to generate launch configuration JSON
+             *         for the adapter based on the selection in the "Attach to Process" dialog, and specify
+             *         its CLSID in the "AdapterLauncher" property below.
+             */
+            engineKey.SetValue("Attach", 1);
+            // engineKey.SetValue("PortSupplier", "{708C1ECA-FF48-11D2-904F-00C04FA302A1}");
+            engineKey.SetValue("AdapterLauncher", _adapterLauncherCLSID);
+
+            /*
+             * Set to "1" if the debug adapter will use the VS "Exception Setting" tool window.  The debug adapter's must
+             * support one of the following:
+             *     -Exception Breakpoints
+             *         The debug adapter's response to the "initialize" request must contain a set of ExceptionBreakpointFilters,
+             *         and the "ExceptionBreakpointCategory" property must be defined below.An optional set of
+             *         "ExceptionBreakpointMappings" may also be provided if the VS exception names do not correspond to the
+             *         "Label" properties of the ExceptionBreakpointFilters.
+             *     -Exception Options
+             *         The debug adapter's response to the "initialize" request must contain the "SupportsExceptionOptions"
+             *         and "SupportsExceptionDetailsRequest" flags, and ExceptionCategoryMapping information must be supplied.
+             */
+            engineKey.SetValue("Exceptions", 1);
+
+            /*
+             * Set to "1" if the debug adapter supports the VS exception conditions experience(For skipping exceptions in specific modules).
+             * The debug adapter's response to the "initialize" request must contain the "SupportsExceptionConditions"
+             * and "SupportsExceptionDetailsRequest" flags.
+             */
+            engineKey.SetValue("ExceptionConditions", 0);
+
+            /*
+             * Debug Adapter Host settings
+             * These settings control the behavior of the Debug Adapter Host
+             *
+             * Name of the debug adapter
+             * This appears in VS in several places.  For example:
+             *     -The "Select Code Type" dialog for choosing which debugger to attach to a process(if Attach is supported)
+             *     -The "Debugging" column of the "Processes" tool window
+             */
+            engineKey.SetValue("Name", _name);
+
+            /*
+             * Path to the debug adapter executable
+             */
+            engineKey.SetValue("Adapter", @"$PackageFolder$\DebugAdapter.exe");
```

Comment:
Do we actually need this, given that we'll have a custom `IAdapterLauncher` that provides the streams directly, without another process?